nistats/first_level_model.py

```diff
@@ -464,23 +503,27 @@ def compute_contrast(self, contrast_def, contrast_name=None,
 
         Returns
         -------
-        output_image : Nifti1Image
+        output : Nifti1Image
             The desired output image
 
         """
         if self.labels_ is None or self.results_ is None:
             raise ValueError('The model has not been fit yet')
 
-        if isinstance(contrast_def, np.ndarray):
+        if isinstance(contrast_def, (np.ndarray, str)):
             con_vals = [contrast_def]
         elif isinstance(contrast_def, (list, tuple)):
             con_vals = contrast_def
-            for cidx, con in enumerate(contrast_def):
-                if not isinstance(con, np.ndarray):
-                    raise ValueError('contrast_def at index %i is not an'
-                                     ' array' % cidx)
         else:
-            raise ValueError('contrast_def must be an array or list of arrays')
+            raise ValueError('contrast_def must be an array or str or list of'
+                             ' (array or str)')
+
+        # Translate formulas to vectors with patsy
+        design_info = DesignInfo(self.design_matrices_[0].columns.tolist())
+        for cidx, con in enumerate(con_vals):
+            if not isinstance(con, np.ndarray):
+                con_vals[cidx] = design_info.linear_constraint(con).coefs
```

Comment:
I will add this as an issue. I think to be tackled after webpage docs with sphinx and circle ci are implemented.